src/zarr/store/common.py

```diff
@@ -75,30 +80,44 @@ def __eq__(self, other: Any) -> bool:
 
 
 async def make_store_path(
-    store_like: StoreLike | None, *, mode: AccessModeLiteral | None = None
+    store_like: StoreLike | None,
+    *,
+    mode: AccessModeLiteral | None = None,
+    storage_options: dict[str, Any] | None = None,
 ) -> StorePath:
+    from zarr.store.remote import RemoteStore  # circular import
+
     if isinstance(store_like, StorePath):
         if mode is not None:
             assert AccessMode.from_literal(mode) == store_like.store.mode
-        return store_like
+        result = store_like
     elif isinstance(store_like, Store):
         if mode is not None:
             assert AccessMode.from_literal(mode) == store_like.mode
         await store_like._ensure_open()
-        return StorePath(store_like)
+        result = StorePath(store_like)
     elif store_like is None:
         if mode is None:
             mode = "w"  # exception to the default mode = 'r'
-        return StorePath(await MemoryStore.open(mode=mode))
+        result = StorePath(await MemoryStore.open(mode=mode))
     elif isinstance(store_like, Path):
-        return StorePath(await LocalStore.open(root=store_like, mode=mode or "r"))
+        result = StorePath(await LocalStore.open(root=store_like, mode=mode or "r"))
     elif isinstance(store_like, str):
-        return StorePath(await LocalStore.open(root=Path(store_like), mode=mode or "r"))
+        storage_options = storage_options or {}
+        fs, _ = fsspec.url_to_fs(store_like, **storage_options)
```

Comment:
A few thoughts here: 

1. It would be nice to avoid creating the `FileSystem` if possible. A lighter approach could attempt to parse the string and look for `://` (and not `file://`).
2. If we do have to create filesystem, it would be nice to reuse it when creating the `RemoteStore`. Something like `RemoteStore.from_filesystem(...)` could be a nice pattern.

Reply:
See below for the way we used to do this. May be worth replicating parts of this

https://github.com/zarr-developers/zarr-python/blob/e1d98cdf3d35e4cb2815912d82a64c1623347c87/zarr/storage.py#L170

Reply:
Just pushed that refactor. It's a bit bigger than I'd like, but the basic version is that `RemoteStores`'s `__init__` matches what it uses internally: you can only given it an async filesystem.

There's a `from_url` and `from_path` that handles the other stuff that was previously in `__init__`.

I removed `RemoteStore._url`. I *think* that that's safe to remove, but I need to actually think through how it interacted with `store.path`